src/wallet/redux/actions/blockchain.ts

```diff
@@ -1,22 +1,33 @@
-import { Wallet } from '../..';
+export type BlockchainDeployAdjudicatorAction = ReturnType<typeof deployAdjudicator>;
```

Comment:
Maybe call this `DeployAdjudicator` to stick with the new action pattern (you'll get the `blockchain` part from the namespace). Potentially even better would be to have something like:
```ts
type DeploymentRequest = ReturnType<typeof deploymentRequest>;
type DeploymentSuccess = ReturnType<typeof deploymentSuccess>;
type DeploymentWrongNetworkFailure = ReturnType<typeof deploymentWrongNetworkFailure>;
type DeploymentMetamaskFailure = ReturnType<typeof deploymentMetamaskFailure>;
type DeploymentFailure = DeploymentWrongNetworkFailure | DeploymentMetamaskFailure;
type DeploymentResponse = DeploymentSuccess | DeploymentFailure;
```

Reply:
Not entirely sure about the `Deployment*Failure` - maybe those are more general error actions. It's still useful to have that `DeploymentFailure` union type though, as you can then check you've covered all possibilities by using [the exhaustive checking pattern](https://github.com/magmo/rps-poc/blob/master/src/wallet/redux/sagas/wallet.ts#L38).